### ISSUE TITLE
Running cypress tests in local break content type file

### ISSUE THREAD

**Issue by Skylsmoi** [tracim/tracim#2127]
## Bug description & symptom

Running cypress tests locally breaks every content of type File

## Diagnostic

Default config of cypress tests uses a different database but doesn't uses a specific folder for content type file (/backend/depot and /backend/previews)

When running cypress tests, it reset theses folder deleting the files associated with local database.

## Expected

Cypress tests should use different folders

## How to reproduce

- Have a local tracim instance running
- create some content type file
- run cypress tests
- go back to your local instance
- content type files are broken

## Impacted components

functionnal_tests

**Comment by Skylsmoi** [comment]
Basically, you only needs to change these 2 config in cypress_test.ini:
``` diff
--- basic_setup.depot_storage_dir = %(here)s/depot
+++ basic_setup.depot_storage_dir = %(here)s/depot_cypress
--- basic_setup.preview_cache_dir = %(here)s/previews
+++ basic_setup.preview_cache_dir = %(here)s/previews_cypress
```
These 2 folders (depot_cypress and previews_cypress) must exists so it might require something todo elsewhere

**Comment by Skylsmoi** [comment]
@PhilippeAccorsi @inkhey @buxx 
What is the best way to ensure the folders are created (or already exists) on a fresh install of Tracim ?
I see two options for now:
- Add those folders to git
  - note that adding empty folders to git is always a bit hacky
- have `setup_default_backend.sh` creating it

**Comment by inkhey** [comment]
> @PhilippeAccorsi @inkhey @buxx @lebouquetin
> What is the best way to ensure the folders are created (or already exists) on a fresh install of Tracim ?
> I see two options for now:
> 
>     * Add those folders to git
>       
>       * note that adding empty folders to git is always a bit hacky
> 
>     * have `setup_default_backend.sh` creating it

Not sure it make sense to create dir only for testing purpose on a script used to launch tracim to be production ready. adding those folder to git isn't nice too.
will be better, to have a specific script to create dir for testing. One solution may be to use similar dir as default used by .test.env file with some kind of script to create those file.

see also this issue : https://github.com/tracim/tracim/issues/2129

**Comment by PhilippeAccorsi** [comment]
@Skylsmoi We have a specific script to install cypress, so we can add this folder creation inside this script. Its better because this folder is just necessary if you want to use cypress.

**Comment by Skylsmoi** [comment]
I agree with both changing backend/cypress_test.ini to use specific folders and create them automatically using setup_functionnal_tests.sh seems the best option to me

**Comment by inkhey** [comment]
I was just trying to solve these kind of issue but the behavior in order to integrate pushpin for cypress, but all the cypress process is really messy.

To make things better we need that :

1. All the file used by cypress should be in a specific dir.
2. reset of database does remove all the dir and rebuilt it with `create_test_storage_dir.sh` script (need to be update to allow parameter to set default dir)
3. folder creation should be runned before running travis with pserve, unless pserve will failed.
4. caldav should also be started ! it should be runned after folder creation.
5. pushpin have to be runned to, to allow test on tlm received. As the docker run only on default port and we do need to run pushpin in host mode, we do currently have no other choice than run cypress test on same port as dev -> 7999. 

We need to take in account that:
- If we decide to put the specific dir for cypress in /tmp , we need to be sure this dir does exist at start for user, needing to rerun setup_functionnal_tests.sh  is not a good idea.
- having to run tracim + tracim caldav + pushpin + cypress is complex to run cypress test, a one command that run all and stop all is better, but not easy.
